File: R/LinePickingPDF.R
#' Get pdf for distance between two random points
#'
#' Give a shape (square, disk, ...) and 
#' parameters defining the shape, this 
#' function will give a the probability density
#' function for the distances
#' between two Poisson distributed points in the space.
#'
#' @param t vector of points to calculate pdf for. 
#' @param problem \itemize{
#'    \item 0 Square, with side length parameters[0]
#'    \item 1 Disk, with radius parameters[0]
#'    \item 2 Hyper-ball, dimension parameters[0], radius parameters[1]
#'    \item 3 Rectangle, side lengths parameters[0], parameters[1]
#'    \item 4 Line, length parameters[0]
#'    \item 5 Cube, side length parameters[0]
#'    \item 6 Sphere, with radius parameters[0]
#'    \item 7 SphereGeodesic, with radius parameters[0]
#'    \item 8 PrismGeodesic, TBA
#' }
#' @param para the parameter necessary to describe 
#' the space given by problem.
#' @return vector of probability density function values for each
#' element in t.
#' @author Eric Parsonage, Matt Roughan, Jono Tuke
#' @export
#' @useDynLib LinePicking
#' @note August 25 2012
#' @examples
#' t <- seq(0,1,l=1000)
#' y <- LinePickingPDF(t=t,problem=0,para=1)
#' plot(t,y,type='l')
LinePickingPDF <-
function(t,problem=0,para){
  n <- length(t)
  tmp <- .C('LinePickingPDF',
            t = as.double(t),
            pdf = as.double(rep(0,n)),
            n = as.integer(n),
            problem = as.integer(problem),
            para = as.double(para),
            Npar = as.integer(length(para)),
            results = as.integer(0),
            error_str = as.character(""))
  if(tmp$results != 0 ){
    stop(tmp$error_str)
  } else {
    return(tmp$pdf)    
  }
}

File: Matlab/LinePickingPDF.m
function g =  LinePickingPDF(t, problem, parameters) 
%
%     Copyright 2012 Eric Parsonage <eric.parsonage@adelaide.edu.au>
%     This program is free software: you can redistribute it and/or modify
%     it under the terms of the GNU General Public License as published by
%     the Free Software Foundation, either version 3 of the License, or
%     (at your option) any later version.
%   
%     This program is distributed in the hope that it will be useful,
%     but WITHOUT ANY WARRANTY; without even the implied warranty of
%     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
%     GNU General Public License for more details.
%
%     You should have received a copy of the GNU General Public License
%     along with this program.  If not, see <http://www.gnu.org/licenses/>.
% 
% LinePickingPDF(t, problem, parameters) computes the PDF of the length of a 
%     randomly chosen line, within some region chosen by the problem. 
%
% INPUTS:
%    t = vector of points on which to calculate the PDF         
%    problem = region type
%              currently supported: 
%                        0: square, with side length parameters[0]
%                        1: disk, with radius parameters[0]
%                        2: hyper-ball, dimension parameters[0], radius parameters[1]
%                        3: rectangle, side lengths parameters[0], parameters[1]
%                        4: line, length parameters[0]
%                        5: cube, side length parameters[0]
%                        6: sphere, with radius parameters[0]
%                        7: sphere geodesic, with radius parameters[0]
%                        8: prism geodesic, with length parameters[0], and perimeter parameters[1]
%
%
%    parameters = vector of parameters for the region
%         
% OUTPUTS:        
%   g(t) = vector of calculated densities      
%         
%
% See Also LinePickingCDF, LinePickingMean, LinePickingVar, LinePickingSupport, LinePickingCheckParameters, LinePickingProblemLookup, LinePickingAllProblems 
g = LinePicking(0, t, problem, parameters);